Libraries/LibELF/DynamicLoader.cpp

```diff
@@ -323,6 +354,31 @@ void DynamicLoader::setup_plt_trampoline()
 #endif
 }
 
+void DynamicLoader::relocate_got_plt()
+{
+    //
+    // At this stage, the .got.plt contains addresses that point back to
+    // the plt. This is a part of the lazy linking mechanism.
+    // The plt was probably loaded somewhere else than the compiler expected,
+    // so we need to relocate the addresses in .got.plt to point at the actual
+    // indtended parts in the plt.
+
+    // TODO: is this actually the way it's supposed to be done?
+    // why doesn't the compiler explicitly generate relocations for this?
```

Comment:
Not sure if this is your TODO or mine, but it seems to be the correct way of getting the offset of _DYNAMIC and  setting up the PLT for the later R_386_JUMP_SLOT relocations